src/Sylius/Bundle/ApiBundle/Resources/config/services.xml

```diff
@@ -76,5 +76,18 @@
         <service class="Sylius\Bundle\ApiBundle\DataTransformer\OrderTokenValueAwareInputDataTransformer" id="sylius_bundle_api.data_transformer.add_item_to_cart_input_data_transformer">
             <tag name="api_platform.data_transformer"/>
         </service>
+
+        <service id="Sylius\Bundle\ApiBundle\Serializer\AddressDenormalizer">
+            <argument type="service" id="serializer.mapping.class_metadata_factory" />
+            <argument type="service" id="serializer.name_converter.metadata_aware" />
+            <argument type="service" id="serializer.property_accessor" />
+            <argument type="service" id="property_info" on-invalid="ignore" />
+            <argument type="service" id="serializer.mapping.class_discriminator_resolver" on-invalid="ignore" />
+            <argument>null</argument>
+            <argument type="collection" />
+            <argument type="string">%sylius.model.address.class%</argument>
+            <argument type="string">Sylius\Component\Core\Model\AddressInterface</argument>
```

Comment:
As it is an `AddressDenormalizer` I would be for not passing an additional argument here, but using an interface directly in a service